live-examples/js-examples/array/array-reduce.js

```diff
@@ -1,10 +1,19 @@
 const array1 = [1, 2, 3, 4];
-const reducer = (previousValue, currentValue) => previousValue + currentValue;
 
 // 1 + 2 + 3 + 4
-console.log(array1.reduce(reducer));
+const result = array1.reduce(
+  (previousValue, currentValue) => previousValue + currentValue
+);
+
+console.log(result);
 // expected output: 10
 
 // 5 + 1 + 2 + 3 + 4
-console.log(array1.reduce(reducer, 5));
+const initialValue = 5;
+const resultWithInitial = array1.reduce(
+  (previousValue, currentValue) => previousValue + currentValue,
+  initialValue
+);
+
+console.log(resultWithInitial);
```

Comment:
These are nice changes. It helps better explain the initial value example quite a lot and the removal of the overly-generic (IMO) `reducer` function lets the reader focus more on what's important. I see opportunity for more clarity, too.

Given the preceding sentence for this code example:
> Perhaps the easiest-to-understand case for reduce() is to return the sum of all the elements in an array:

The reader already has the word "sum" in mind but the code only connects to that word via the `+` operator.

WDYT of changing `result` to `sum` and `resultWithInitial` to `sumWithInitialValue`?

I also think it's hard for new learners to make the connection between the callback arguments and the array itself. It _might_ also help to connect the dots between the `previousValue` and `currentValue` arguments to the actual array if the array were named `values`. WDYT?

Thanks for working to improve this!

Reply:
I love the sum variable naming idea. I’ll update this PR.